otterai/indexer.py

```diff
@@ -42,59 +45,78 @@ def index_codebase(root_dir: str) -> Dict[str, List[str]]:
         'test': [],
         'other': []
     }
+
+    default_ignore_patterns = [
```

Comment:
🧹 The list of default ignore patterns contains duplicates (e.g., '*.pyc' and '__pycache__/*'). Remove duplicates for better maintainability.